### PR TITLE
feat(auth): add handling for oauth token skew time

### LINKED ISSUE #38: Implement Skew Time for OAuth Token Expiry
This ticket proposes a modification to the `is_token_expired` method to incorporate skew time for the received OAuth token. The current implementation simply compares the `token expiry` value to the current timestamp to determine the token expiry.

### The Need for Skew Time

Clocks on different servers can have slight discrepancies, even when synchronized. This can lead to issues where a client attempts to use a token shortly before its calculated expiry on the client-side, but the token is already expired on the server-side. Implementing skew time adds a buffer to the calculated expiry, mitigating this risk.

### Proposed Change

The change involves introducing a skew time parameter to the `is_token_expired` method. This parameter will represent a time offset (likely in seconds) that will be subtracted from the expiry time based on the provided `expires_in` value.

### PR DESCRIPTION
**Pull request by MaryamAdnan3**
## What
This PR updates the `is_token_expired` utility method to minimize the impact of clock skew between client and server during OAuth token validation.

## Why
1. Honors provided skew buffer.
3. Accounts for round-trip time for a more accurate expiry calculation. Improves token validation reliability and reduces unnecessary token refreshes.

Closes #38 

## Type of change
Select multiple if applicable.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause a breaking change)
- [ ] Tests (adds or updates tests)
- [ ] Documentation (adds or updates documentation)
- [ ] Refactor (style improvements, performance improvements, code refactoring)
- [ ] Revert (reverts a commit)
- [ ] CI/Build (adds or updates a script, change in external dependencies)

## Dependency Change
If a new dependency is being added, please ensure that it adheres to the following guideline https://github.com/apimatic/apimatic-codegen/wiki/Policy-of-adding-new-dependencies-in-the-core-libraries

## Breaking change
If the PR is introducing a breaking change, please ensure that it adheres to the following guideline https://github.com/apimatic/apimatic-codegen/wiki/Guidelines-for-maintaining-core-libraries

## Testing
List the steps that were taken to test the changes

## Checklist
- [ ] My code follows the coding conventions
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added new unit tests

### CODE REVIEW

File: lib/apimatic-core/utilities/auth_helper.rb [hunk 1]
```diff
@@ -17,9 +17,10 @@ def self.get_base64_encoded_value(*props, delimiter: ':')
     # Checks if OAuth token has expired.
     # @param [int] token_expiry The expiring of a token.
     # @return [Boolean] true if token has expired, false otherwise.
-    def self.token_expired?(token_expiry)
+    def self.token_expired?(token_expiry, clock_skew_time = nil)
```
Comment (sufyankhanrao) [inline lib/apimatic-core/utilities/auth_helper.rb:20]: Non-blocker: The method doc should be updated with the newly added parameter.

File: test/test-apimatic-core/utilities/auth_helper_test.rb [hunk 2]
```diff
@@ -52,6 +52,31 @@ def test_is_token_expired_false
...
+  def test_token_is_expired_with_clock_skew
+    token_expiry = Time.now().utc.to_i + 3600
+    actual_value = AuthHelper.token_expired?(token_expiry, 7200)
+    expected_value = true
+
+    refute_nil actual_value
+    assert_equal expected_value, actual_value
+  end
+
+  def test_token_is_not_expired_with_clock_skew
```
Comment (sufyankhanrao) [inline test/test-apimatic-core/utilities/auth_helper_test.rb:64]: Blocker: There should be another test case for a negative clock_skew value so that it helps in avoiding any broken logic in future.

**Review by sufyankhanrao — CHANGES_REQUESTED** [review]

**Review by sufyankhanrao — APPROVED** [review]
